python/rmm/_lib/device_buffer.pyx

```diff
@@ -92,39 +95,42 @@ cdef class DeviceBuffer:
         return buf
 
     @staticmethod
-    @cython.boundscheck(False)
-    cdef DeviceBuffer c_frombytes(const unsigned char[::1] b,
+    cdef DeviceBuffer c_to_device(const unsigned char[::1] b,
                                   uintptr_t stream=0):
-        if b is None:
-            raise TypeError(
-                "Argument 'b' has incorrect type"
-                " (expected bytes, got NoneType)"
+        return to_device(b, stream)
+
+    @staticmethod
+    def to_device(const unsigned char[::1] b, uintptr_t stream=0):
+        return to_device(b, stream)
+
+    cpdef copy_to_host(self, unsigned char[::1] hb=None, uintptr_t stream=0):
+        cdef const device_buffer* dbp = self.c_obj.get()
+        cdef size_t s = dbp.size()
+
+        if hb is None:
+            # NumPy leverages huge pages under-the-hood,
+            # which speeds up the copy from device to host.
+            hb = np.empty((s,), dtype="u1")
+        elif len(hb) < s:
+            raise ValueError(
+                "Argument `hb` is too small. Need space for %i bytes." % s
             )
+        elif len(hb) > s:
+            hb = hb[:s]
```

Comment:
Offline Keith pointed out that this results in returning a small view of the buffer instead of just the original buffer. That seems weird for a user. So we shouldn't perform the assignment here, but just pass the selection below. Will update accordingly.

Reply:
Addressed in the commit below